src/Examples/MatOpt/nanowire_design.ipynb

```diff
@@ -0,0 +1,441 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "canvas = Canvas.fromLatticeAndShape(lattice, shape)\n",
+    "canvas.makePeriodic(tiling, lattice.getNeighbors)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `canvas` object holds information about the design space and the lattice sites, but it does not specify any material building block information. In this example, we are considering an InAs nanowire. To represent material configurations, we will use a **Design** object. We first construct an empty Design from `canvas` and then place **Atom** (i.e., building block) objects into this Design via methods provided by `lattice`.\n",
+    "\n",
+    "To debug our work so far, we can create material structure files to load and plot with standard visualization tools. Here, we create a PDB (protein data bank format, www.rcsb.org) file for the Design object. These files can be plotted with visualization packages such as AtomEye or OVITO."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "design = Design(canvas)\n",
+    "lattice.setDesign(design, Atom('In'), Atom('As'))\n",
+    "design.toPDB('canvas.pdb')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Building a Model\n",
+    "\n",
+    "In this example, we will build a model that maximizes the cohesive energy of the nanowire as an indicator of it's thermal stability. To begin, we start by creating a **MatOptModel** object to hold information about the model. Notice that we use a list of empty atoms as our set of building blocks because we will use a cohesive energy function for semiconductor materials that is indepedent of atom types. Thus, we do not need to specify the types of building blocks in our model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = MatOptModel(canvas, [Atom('')])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By default, several basic variables are pre-defined. See the first example, ***Monometallic_Nanocluster_Design.ipynb*** for a description of basic variables, expressions, and constraint rules."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we fix the core layers of the nanowire specified by the ratio of the core radius and the canvas radius."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "coreRatio = 0.2\n",
+    "CoreLayers = [i for i, p in enumerate(canvas.Points) if p[0] ** 2 + p[1] ** 2 < (coreRatio * radius) ** 2]\n",
+    "m.Yi.rules.append(FixedTo(1, sites=CoreLayers))"
```

Comment:
I know this is an advanced MatOpt example, but it may be useful to just remind the reader/user what the .rules member of the basic variables is for and what adding to it is doing in terms of framing the model.